lib/vc.js

```diff
@@ -82,24 +90,55 @@ async function issue(options = {}) {
   });
 
   // run common credential checks
-  await _checkCredential({credential: options.credential});
+  const {credential} = options;
+  if(!credential) {
+    throw new TypeError('"credential" property is required for issuing.')
+  }
+  _checkCredential(credential);
 
-  return jsigs.sign(options.credential, {purpose, documentLoader, ...options});
+  return jsigs.sign(credential, {purpose, documentLoader, ...options});
 };
 
 /**
- * Verify a credential.
+ * Verifies a credential or presentation.
+ *
+ * @param {object} [options={}]
+ *
+ * Required, one of:
+ * @param {object} [options.credential] - Verifiable credential
+ * @param {object} [options.presentation] - Verifiable Presentation
+ * @throws {TypeError} If neither credential or presentation is passed.
+ *
+ * @param {CredentialIssuancePurpose} [options.purpose]
+ * @param {function} [options.documentLoader]
+ *
+ * @returns {Promise<{verified: boolean, results: *[], error: Error}>}
  */
 async function verify(options = {}) {
-  const {credential} = options;
+  const {credential, presentation} = options;
+
+  if(!credential && !presentation) {
+    throw new TypeError(
+      'A "credential" or "presentation" property is required for verifying.')
+  }
+
+  // check if developer mistakenly passed in a VP as 'credential'
+  if(credential && _isPresentation(credential)) {
+    throw new TypeError('Use "presentation" property when verifying VPs.')
+  }
+
   // run common credential checks
   try {
```

Comment:
This might be easier to follow by moving the try and cred logic to a _verifyCredential which is similar to the _verifyPresentation function.  It seems a bit mixed up here.  Then this function would just have:
```js
if(credential) {
  return _verifyCredential(options);
} else {
  return _verifyPresentation(options);
}
```
(can probably just pass options directly since those functions would just ignore the other option)

Reply:
Yeah, good call. Changing.

Reply:
Ok, I let me know if the latest changes reflect your request.